packages/@mantine/core/src/core/styles-api/use-styles/use-styles.ts

```diff
@@ -36,6 +37,18 @@ export type GetStylesApi<Payload extends FactoryPayload> = (
   style: CSSProperties;
 };
 
+// Omit objects from props to avoid unnecessary rerenders
+function omitObjectProps(props: Record<string, any>): Record<string, any> {
+  const result: Record<string, any> = {};
+  for (const [key, value] of Object.entries(props)) {
+    if (!value || (typeof value !== 'object' && typeof value !== 'function')) {
+      result[key] = props[key];
+    }
+  }
+
+  return result;
+}
```

Comment:
How about using Record<string, unknown>? It helps to improve type safety compared to any.

Also, is there a reason you used props[key]? Would it be problematic to just use value?